### PR TITLE
Rewrite daily update to run under Argo

### PR DESCRIPTION
**Pull request by moseshll**
Changes made to `exe/hathifiles_database_update` in the spirit of those made to `exe/hathifiles_database_full` under `DEV-1087_argo_monthly`

### CODE REVIEW

**Review by aelkiss — COMMENTED** [review]
Overall looks fine so long as we keep track of the need at some point to address date-independence here. I do want to replace `$:` with `$INCLUDE_PATH` though.

File: exe/hathifiles_database_update [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-connection_string = ENV['HATHIFILES_MYSQL_CONNECTION'] || prompt.ask("Connection string:")
-connection = HathifilesDatabase.new(connection_string)
+envfile = Pathname.new(__dir__).parent + ".env"
+Dotenv.load(envfile)
 
-connection.logger.info "Connecting to #{connection_string}"
+# Use the daily hathifile from command line
+hathifile = ARGV[0]
+# Fall back to the most recent daily hathifile if not specified
+if hathifile.nil?
```
[the remark this answers is not in the document]
Comment (moseshll) [reply exe/hathifiles_database_update:19]: I am not aware of any flags getting set, or latest filename recorded. In this case there's nothing to indicate the previous state. I would suggest a database table as an interim solution, that would buy us date independence fairly cheaply.

File: exe/hathifiles_database_update [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
-Dotenv.load(envfile)
+#!/usr/bin/env ruby
 
-filename = ARGV[0]
-tempdir = Pathname.new('.').realdirpath + 'tmp'
+# Invoke Connection class to load a hathifile, using the most recent one if
+# not specified on the command line.
 
-prompt = TTY::Prompt.new
+$:.unshift "../lib"
```
[the remark this answers is not in the document]
Comment (moseshll) [reply exe/hathifiles_database_update]: I'm a little surprised `standardrb` doesn't complain about it.

File: exe/hathifiles_database_update (hunk 1 above)
[the remark this answers is not in the document]
Comment (aelkiss) [reply exe/hathifiles_database_update:19]: That would work; again, I don't think any need to do it right now.

**Pull request merged**